src/output_writers.jl

```diff
@@ -324,3 +324,72 @@ function time_to_write(clock, out::JLD2OutputWriter)
         return false
     end
 end
+
+mutable struct Checkpointer <: OutputWriter
+                 dir :: String
+              prefix :: String
+    output_frequency :: Int
+end
+
+function Checkpointer(; output_frequency, dir=".", prefix="checkpoint", force=false)
+    mkpath(dir)
+    return Checkpointer(dir, prefix, output_frequency)
+end
+
+function savesubfields!(file, model, name, flds=propertynames(getproperty(model, name)))
+    for f in flds
+        file["$name/$f"] = Array(getproperty(getproperty(model, name), f).data.parent)
+    end
+    return nothing
+end
+
+checkpointed_structs   = [:arch, :boundary_conditions, :grid, :clock, :eos, :constants, :closure]
```

Comment:
These fields could be made properties of the `Checkpointer`. This will allow the user to add/remove fields from checkpointing, which may be generally useful in the future and is currently useful for omitting boundary conditions from checkpointing.

Reply:
@ali-ramadhan was this resolved?